src/client_builder.rs

```diff
@@ -61,41 +155,132 @@ where
         ClientBuilder {
             domain,
             port,
-            starttls: false,
+            mode: ConnectionMode::Auto,
```

Comment:
I'm tempted to have `mode` default to `ConnectionMode::Tls`, just because these days it really is what most sane clients should be using (even `STARTTLS` has its problems). Users can then always override that behavior if they know they want to connect to a less-secure server. Maybe we have a helper method called `.compatible()` or something that will set `Auto + Any`?

Reply:
Most of my clients do starttls by default vs using the SSL only port.   And anyways if they connect to port 993 it'll use Tls full anyway in Auto mode.

Maybe add a new mode that is AutoTls that will connect to any Tls (starttls or tls) by default only (and not non-tls).. and if the user wants to opt-in to anything (even insecure) they can just set it to auto explicitly?